cora/util/cubicspline.pyx

```diff
@@ -191,21 +188,13 @@ cdef class Interpolater(object):
         cdef Py_ssize_t length = n - 2
         cdef Py_ssize_t i
 
-        al_raw = np.zeros(length, dtype=np.float64)
-        bt_raw = np.zeros(length, dtype=np.float64)
-        gm_raw = np.zeros(length, dtype=np.float64)
-        f_raw = np.zeros(length, dtype=np.float64)
-        z_raw = np.zeros(length, dtype=np.float64)
-        l_raw = np.zeros(length, dtype=np.float64)
-        m_raw = np.zeros(length, dtype=np.float64)
-
-        cdef double[:] al = al_raw
-        cdef double[:] bt = bt_raw
-        cdef double[:] gm = gm_raw
-        cdef double[:] f = f_raw 
-        cdef double[:] z = z_raw 
-        cdef double[:] l = l_raw 
-        cdef double[:] m = m_raw 
+        cdef double[:] al = np.zeros(length, dtype=np.float64)
+        cdef double[:] bt = np.zeros(length, dtype=np.float64)
+        cdef double[:] gm = np.zeros(length, dtype=np.float64)
+        cdef double[:] f = np.zeros(length, dtype=np.float64)
+        cdef double[:] z = np.zeros(length, dtype=np.float64)
+        cdef double[:] l = np.zeros(length, dtype=np.float64)
+        cdef double[:] m = np.zeros(length, dtype=np.float64)
```

Comment:
It's always advantageous to mark arrays as contiguous if you know that they are.
```suggestion
        cdef double[::1] al = np.zeros(length, dtype=np.float64)
        cdef double[::1] bt = np.zeros(length, dtype=np.float64)
        cdef double[::1] gm = np.zeros(length, dtype=np.float64)
        cdef double[::1] f = np.zeros(length, dtype=np.float64)
        cdef double[::1] z = np.zeros(length, dtype=np.float64)
        cdef double[::1] l = np.zeros(length, dtype=np.float64)
        cdef double[::1] m = np.zeros(length, dtype=np.float64)
```